content/news/XXXX-XX-XX-jack-zero-latency.md

```diff
@@ -0,0 +1,77 @@
+The round trip latency is 95 ms (driver + ALSA + JACK async mode + duplex stream = 4 buffers).
+
+The lower stream is the ALSA case. Mixxx is configures with the same single buffer of 1024 frames = 23.2 ms
+The round trip latency is 49 ms (driver + ALSA = 2 buffers).
+
+Not in the picture is the ALSA pulse device. It runs at a latency of 104 ms (driver + ALSA + 2 x pulse + ALSA = 5 buffers)
+PipeWire on Fedora 34 has by default the same latency as JACK in this test.
+
+With this picture we can verify that Mixxx actually has the same buffer size in both cases. When pressing pause, it fades the signal out over one buffer length which is equal in both cases.
+The peaks in the recorded right channel is the sound of the mouse click. You can only barely see the recorded sine wave.
+
+![Screenshot of audacity showing the fade out]({static}/images/news/fadeoutcompare.png)
+
+For reference, I have done the same test using [`jack_iodelay`](http://manpages.ubuntu.com/manpages/bionic/man1/jack_iodelay.1.html)
+
+
+    3114.842 frames     70.631 ms total roundtrip latency
+    extra loopback latency: 42 frames
+    use 21 for the backend arguments -I and -O
+
+
+The result is 70 ms (driver + ALSA + JACK async mode = 3 buffers). The duplex cycle is omitted here. This is one buffer more than a native ALSA implementation.
+
+To verify that the duplex cycle is not introduced by Mixxx or the [PortAudio](http://www.portaudio.com) abstraction layer used by Mixxx, I have routed the `jack_iodelay` signal through Mixxx.
+
+![Jack patch field showing the jack_iodelay Mixxx loop]({static}/images/news/jackpatch.png)
+
+    2048.000 frames     46.440 ms total roundtrip latency
+    extra loopback latency: 2047 frames
+    use 1023 for the backend arguments -I and -O
+
+This is the minimum we can expect, one buffer is needed for `jack_iodelay` and one for Mixxx. The result can be confirmed with the same setup using `jack_latent_client` which just passes the input to the output.
+
+If we connect the output of Mixxx to the input, an extra buffer is used for some reason. This can be also confirmed with `jack_latent_client` and the same feedback loop.
+
+![Jack feedback loop]({static}/images/news/feedbackloop.png)
+
+The upper stream shows the Mixxx results and the lower stream shows the `jack_latent_client` results.
+
+In addition, most distros use JACK with it's default "Server Asynchronous Mode" which introduce one buffer extra latency (`--async-latency`) when accessing the sound card. From Ubuntu Hirsute 21.4, QJackCtl exposes a "Use server synchronous mode" checkbox in the sound card preferences. It is grayed out by default but becomes active if "Enable JACK D-Bus interface" is checked as well. In this case the JACK mixing is done after Mixxx in the same time interval.
+
+## Conclusion
+
+For now, we recommend using Mixxx with the ALSA backends even if you are running JACK. The same applies to [PipeWire](https://pipewire.org) as Mixxx uses JACK protocol to connect to it.
```

Comment:
I would not explicit recommend it, because I think it is not default for some reasons. It is obvious that Jack needs some time to compose the final buffer and I assume that this time is removed from the time Mixxx can use. 

It might be the case that you can use in Async mode a smaller buffer, resulting to the same over all latency with a bigger buffer in sync mode, but more predictable. 

At least we need to collect some experience before recommending it.